tests/models/encodec/test_modeling_encodec.py

```diff
@@ -142,6 +146,213 @@ class EncodecModelTest(ModelTesterMixin, PipelineTesterMixin, unittest.TestCase)
     test_resize_embeddings = False
     pipeline_model_mapping = {"feature-extraction": EncodecModel} if is_torch_available() else {}
 
+    # Test copied from: https://github.com/facebookresearch/encodec/blob/main/encodec/msstftd.py#L132
+    @slow
+    def test_discriminator_output_shapes_and_feature_maps(self):
+        disc = EncodecDiscriminator(EncodecDiscriminatorConfig())
+        y = torch.randn(1, 1, 24000)
+        y_hat = torch.randn(1, 1, 24000)
+
+        y_disc_r, fmap_r = disc(y)
+        y_disc_gen, fmap_gen = disc(y_hat)
+        assert len(y_disc_r) == len(y_disc_gen) == len(fmap_r) == len(fmap_gen) == disc.num_discriminators
+
+        assert all(len(fm) == 5 for fm in fmap_r + fmap_gen)
+        assert all(list(f.shape)[:2] == [1, 32] for fm in fmap_r + fmap_gen for f in fm)
+        assert all(len(logits.shape) == 4 for logits in y_disc_r + y_disc_gen)
```

Comment:
I think we'd rather use somethings like `self.assertTrue`, `self.assertListEqual` and `self.assertEqual` instead of `assert` !